astropy/units/tests/test_quantity_ufuncs.py

```diff
@@ -296,6 +296,18 @@ def test_power_array_array(self):
     def test_power_array_array2(self):
         np.power([2., 4.] * u.m, [2., 4.])
 
+    def test_power_array_array3(self):
+        # Equivalent unit fractions are NOT considered dimensionless and raise
+        # a ValueError.
+        with raises(ValueError):
+            np.power([2., 4.] * u.m / u.cm, [2., 4.])
+
+        # But identical unit fractions are considered dimensionless and are
+        # allowed as base for np.power: #4764
+        res = np.power([2., 4.] * u.m / u.m, [2., 4.])
```

Comment:
@pllim I totally forgot to include it but that wasn't part of the problem and with the change both of these will work:
```
np.power(np.array([2, 2, 2, 2]) * u.Unit('cm/cm'), np.array([1, 2, 3, 4]) * u.Unit('')) 
np.power(np.array([2, 2, 2, 2]) * u.Unit('cm/cm'), np.array([1, 2, 3, 4]) * u.Unit('m/m'))
```

Of course I can include them if you want. :)